File: LeetCode/Medium/11 ContainerWithMostWater/11ContainerWithMostWater.py
from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1
        ans = 0
        while l < r:
            ancho = r - l
            alto = min(height[r], height[l])
            area = ancho * alto
            ans = max(ans, area)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return ans

if __name__ == "__main__":
    sol = Solution()
    alturas = [1, 8, 6, 2, 5, 4, 8, 3, 7]
    resultado = sol.maxArea(alturas)
    print(resultado)

File: LeetCode/Medium/11 ContainerWithMostWater/11ContainerWithMostWater.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0;
        int r = height.size() - 1;
        int ans = 0;
        while (l < r) {
            int ancho = r - l;
            int alto = min(height[r], height[l]);
            int area = ancho * alto;
            ans = max(ans, area);
            if (height[l] < height[r]) {
                l++;
            } else {
                r--;
            }
        }
        return ans;
    }
};

int main() {
    Solution sol;
    vector<int> height = {1, 8, 6, 2, 5, 4, 8, 3, 7};
    int resultado = sol.maxArea(height);
    cout << resultado << endl;
    return 0;
}